In [25]:
import pandas as pd
import numpy as np




def genMat():
    return np.array([[np.random.randint(-10, 11),np.random.randint(-10, 11),np.random.randint(-10, 11)],[np.random.randint(-10, 11),np.random.randint(-10, 11),np.random.randint(-10, 11)],[np.random.randint(-10, 11),np.random.randint(-10, 11),np.random.randint(-10, 11)]])

A = genMat()
B = genMat()
C = genMat()

Ainv = np.linalg.inv(A)
Binv = np.linalg.inv(B)
Cinv = np.linalg.inv(C)

print(A@Ainv)
print(B@Binv)
print(C@Cinv)
# Question 4.2: You can see the inverse time the matrix equals the identity matrix.Question 4.2: You can see the inverse time the matrix equals the identity matrix.


[[ 1.00000000e+00  2.77555756e-17 -1.11022302e-16]
 [-1.66533454e-16  1.00000000e+00  1.66533454e-16]
 [ 2.77555756e-16  2.77555756e-17  1.00000000e+00]]
[[ 1.00000000e+00  1.11022302e-16  2.22044605e-16]
 [ 0.00000000e+00  1.00000000e+00  0.00000000e+00]
 [ 1.38777878e-17 -1.11022302e-16  1.00000000e+00]]
[[ 1.00000000e+00 -4.44089210e-16  0.00000000e+00]
 [ 0.00000000e+00  1.00000000e+00  0.00000000e+00]
 [-4.44089210e-16 -2.22044605e-16  1.00000000e+00]]


Question 4.2: You can see the inverse time the matrix equals the identity matrix.Question 4.2: You can see the inverse time the matrix equals the identity matrix.


In [26]:
data = pd.read_csv("kc_house_data.csv")

print(data.columns)



Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')


In [27]:
data.drop(columns = ['id','date'],inplace=True)
data

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0      221900.0         3       1.00         1180      5650     1.0   
1      538000.0         3       2.25         2570      7242     2.0   
2      180000.0         2       1.00          770     10000     1.0   
3      604000.0         4       3.00         1960      5000     1.0   
4      510000.0         3       2.00         1680      8080     1.0   
...         ...       ...        ...          ...       ...     ...   
21608  360000.0         3       2.50         1530      1131     3.0   
21609  400000.0         4       2.50         2310      5813     2.0   
21610  402101.0         2       0.75         1020      1350     2.0   
21611  400000.0         3       2.50         1600      2388     2.0   
21612  325000.0         2       0.75         1020      1076     2.0   

       waterfront  view  condition  grade  sqft_above  sqft_basement  \
0               0     0          3      7        1180              0   
1               0     0          3      7        2170            400   
2               0     0          3      6         770              0   
3               0     0          5      7        1050            910   
4               0     0          3      8        1680              0   
...           ...   ...        ...    ...         ...            ...   
21608           0     0          3      8        1530              0   
21609           0     0          3      8        2310              0   
21610           0     0          3      7        1020              0   
21611           0     0          3      8        1600              0   
21612           0     0          3      7        1020              0   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0          1955             0    98178  47.5112 -122.257           1340   
1          1951          1991    98125  47.7210 -122.319           1690   
2          1933             0    98028  47.7379 -122.233           2720   
3          1965             0    98136  47.5208 -122.393           1360   
4          1987             0    98074  47.6168 -122.045           1800   
...         ...           ...      ...      ...      ...            ...   
21608      2009             0    98103  47.6993 -122.346           1530   
21609      2014             0    98146  47.5107 -122.362           1830   
21610      2009             0    98144  47.5944 -122.299           1020   
21611      2004             0    98027  47.5345 -122.069           1410   
21612      2008             0    98144  47.5941 -122.299           1020   

       sqft_lot15  
0            5650  
1            7639  
2            8062  
3            5000  
4            7503  
...           ...  
21608        1509  
21609        7200  
21610        2007  
21611        1287  
21612        1357  

[21613 rows x 19 columns]

In [28]:
data.agg(['min', 'max', 'mean', 'var'])

price   bedrooms  bathrooms    sqft_living      sqft_lot  \
min   7.500000e+04   0.000000   0.000000     290.000000  5.200000e+02   
max   7.700000e+06  33.000000   8.000000   13540.000000  1.651359e+06   
mean  5.400881e+05   3.370842   2.114757    2079.899736  1.510697e+04   
var   1.347824e+11   0.865015   0.593151  843533.681368  1.715659e+09   

        floors  waterfront      view  condition      grade     sqft_above  \
min   1.000000    0.000000  0.000000   1.000000   1.000000     290.000000   
max   3.500000    1.000000  4.000000   5.000000  13.000000    9410.000000   
mean  1.494309    0.007542  0.234303   3.409430   7.656873    1788.390691   
var   0.291588    0.007485  0.587243   0.423467   1.381703  685734.667269   

      sqft_basement     yr_built   yr_renovated       zipcode        lat  \
min        0.000000  1900.000000       0.000000  98001.000000  47.155900   
max     4820.000000  2015.000000    2015.000000  98199.000000  47.777600   
mean     291.509045  1971.005136      84.402258  98077.939805  47.560053   
var   195872.668401   862.797262  161346.211862   2862.787835   0.019200   

            long  sqft_living15    sqft_lot15  
min  -122.519000     399.000000  6.510000e+02  
max  -121.315000    6210.000000  8.712000e+05  
mean -122.213896    1986.552492  1.276846e+04  
var     0.019833  469761.239945  7.455182e+08

Question 5.1

Lowest Avg:
Long: -121.315

Highest Avg:
zipcode: 90877.939805

Lowest Vari:
waterfront: 0.007485

Highest Vari:
price: 1.347824e+11







In [29]:
response = 'price'
corr = data.corr(numeric_only=True)[response]
corr_table = corr.drop(response).to_frame(name='correlation')
corr_table

correlation
bedrooms          0.308350
bathrooms         0.525138
sqft_living       0.702035
sqft_lot          0.089661
floors            0.256794
waterfront        0.266369
view              0.397293
condition         0.036362
grade             0.667434
sqft_above        0.605567
sqft_basement     0.323816
yr_built          0.054012
yr_renovated      0.126434
zipcode          -0.053203
lat               0.307003
long              0.021626
sqft_living15     0.585379
sqft_lot15        0.082447

Question 5.2: sqfr_living have highest correlation coefficient when the response is price.

Question 5.3: zipcode has a negative correlation coefficient when the response is price.